snapshot-labs/snapshot
# New oSnap features

### Issues
*_(Include references to issues that this PR addresses)_*

Fixes #

Fixes #

### Changes 
*_(Briefly describe the changes made in this PR)_*

1. Updates the oSnap module integration to use the updated OptimisticGovernor contract, which targets our latest oracle contract (OptimisticOracleV3) and has been audited by OpenZeppelin.
2. The new module contract automatically deletes disputed proposals with a callback function, so the SafeSnap plug-in has been updated to reflect that by removing the manual proposal deletion flow.
3. There are additional improvements in how we track a proposal through its lifecycle, since the updated OptimisticGovernor contract tracks the `assertionId` in the OOv3 contract.

### How to test
*_(Explain how the changes can be tested, including any required setup steps)_*

1. Create a testnet Snapshot space, or use an existing one, that has a SafeSnap instance.
2. Manually deploy an oSnap module with the relevant Safe. There will be a Zodiac pull request created later today that makes deployment through the Zodiac app easy. We can help you with manual deployments in the meantime, or you can use our outcome.eth Snapshot space, which is already configured (https://snapshot.org/#/outcomefinance.eth).
3. Run `yarn dev` to create a localhost Snapshot interface.
4. Propose and execute transactions through the Snapshot interface.
5. To test the dispute flow, for the moment you can manually dispute through the OOv3 contract on Etherscan. (https://goerli.etherscan.io/address/0x9923D42eF695B5dd9911D05Ac944d4cAca3c4EAB#code) UMA's new oracle dapp that supports OOv3 will have its production launch in early or middle next week, and we expect to have a demo version ready for you to use this Wednesday, which will let you dispute through the oracle dapp rather than manually.

If you run into problems in testing, please let me know and I will jump to help you. I know there are a few moving pieces, with the Zodiac pull request and the new oracle dapp release. We're really excited about the latest oSnap updates and hope to get this in before the off-site!

Extra note: Since we don't have any production deployments using the old version of the oSnap contract, this pull request is not backwards compatible. It's much simpler to only support the new version in the Snapshot interface. If there are any projects that already have testnet oSnap deployments using the old contract, it will be easy for them to deploy a new module and migrate once the new Zodiac release is out. 

### To-Do
*_(List any outstanding tasks be addressed before or after this PR is merged)_*



### Self-review checklist

### Additional notes or considerations
*_(Include any other relevant information or context that may be helpful for the reviewer)_*

Since there will be a transitional period of about a week and a half until the new oracle dapp is ready, UMA will make public comms, reach out to known oSnap deployers, and make a temporary note in our docs so that partners don't get confused. If the review is positive, we don't need to wait to merge the Snapshot pull request, even if it comes before the new Zodiac release and oracle dapp release. This pull request should only affect very early oSnap users and UMA can coordinate with them.
